src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -1793,6 +1794,19 @@ class TaskProcessor {
                     contextMap.put( param.name, val )
                     break
 
+                case FilemapInParam:
+                    def normalized = normalizeInputToFiles(val.values(),count)
+                    def mapfile = val
+                    int i = 0
+                    mapfile.keySet().each {
+                        mapfile[it] = Paths.get(normalized[i++].stageName)
+                    }
+
+                    contextMap.put( param.name,  mapfile)
+                    count += normalized.size()
+                    task.setInput(param, normalized)
+                    return
+
```

Comment:
Probably it doesn't. Where are S3 files resolved? In order to put together metadata and files, would it be good to have to have `makeTaskContextStage1` processing the metadata and keep the rest for `makeContextStage2` ? This would go along the lines of how it is already implemented.